### PR TITLE
feat: adding support for Buffer.isBuffer and Buffer.isEncoding

### PR DESCRIPTION
**Pull request by lewisgcm**
### Issue # (if available)

No issue associated with this change.

### Description of changes
This PR adds support for `Buffer.isBuffer` and `Buffer.isEncoding` (see https://nodejs.org/api/buffer.html).

### Checklist

- [Y] Created unit tests in `tests/unit` and/or in Rust for my feature if needed
- [Y] Ran `make fix` to format JS and apply Clippy auto fixes
- [Y] Made sure my code didn't add any additional warnings: `make check`
- [NA] Added relevant type info in `types/` directory
- [NA] Updated documentation if needed ([API.md](API.md)/[README.md](README.md)/Other)

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: modules/llrt_buffer/src/buffer.rs [hunk 1]
```diff
@@ -120,6 +120,25 @@ fn byte_length<'js>(ctx: Ctx<'js>, value: Value<'js>, encoding: Opt<String>) ->
     ))
 }
 
+fn is_buffer<'js>(ctx: Ctx<'js>, value: Value<'js>) -> Result<bool> {
+    let class: Function = ctx.globals().get(stringify!(Buffer))?;
```
Comment (richarddavison) [inline modules/llrt_buffer/src/buffer.rs:124]: Use primordials here:
```rust
let constructor = BufferPrimordials::get(ctx)?.consturctor
```

File: modules/llrt_buffer/src/buffer.rs [hunk 2]
```diff
@@ -120,6 +120,25 @@ fn byte_length<'js>(ctx: Ctx<'js>, value: Value<'js>, encoding: Opt<String>) ->
...
+    if let Some(object) = value.as_object() {
+        return Ok(object.is_instance_of(class));
+    }
+
+    Ok(false)
+}
+
+fn is_encoding(value: Value) -> Result<bool> {
+    if let Some(js_string) = value.as_string() {
+        if let Ok(std_string) = js_string.to_string() {
```
Comment (richarddavison) [inline modules/llrt_buffer/src/buffer.rs:134]: We should fail here if we can't convert the JsString to rust string
```rust
let string = js_string.to_string()?;
```

**Review by richarddavison — CHANGES_REQUESTED** [review]
Thanks, some minor comments!

File: modules/llrt_buffer/src/buffer.rs (hunk 1 above)
Comment (lewisgcm) [reply modules/llrt_buffer/src/buffer.rs]: Thanks, i've addressed this now in the next commit.

File: modules/llrt_buffer/src/buffer.rs (hunk 2 above)
Comment (lewisgcm) [reply modules/llrt_buffer/src/buffer.rs]: Good catch, addressed in next commit

**Review by richarddavison — APPROVED** [review]

**Pull request merged**